pkg/graphviz/traceflow_test.go

```diff
@@ -181,6 +181,139 @@ func TestGenGraph(t *testing.T) {
 		},
 	}
 
+	liveTfwithNoSourcePod := crdv1alpha1.Traceflow{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-live-traceflow-nosourcepod",
+		},
+		Spec: crdv1alpha1.TraceflowSpec{
+			LiveTraffic: true,
+			Packet: crdv1alpha1.Packet{
+				IPHeader: crdv1alpha1.IPHeader{
+					Protocol: 1,
+				},
+				TransportHeader: crdv1alpha1.TransportHeader{
+					ICMP: &crdv1alpha1.ICMPEchoRequestHeader{
+						ID:       123,
+						Sequence: 1,
+					},
+				},
+			},
+			Source: crdv1alpha1.Source{
+				IP: "192.168.225.5",
+			},
+		},
+		Status: crdv1alpha1.TraceflowStatus{
+			CapturedPacket: &crdv1alpha1.Packet{
+				DstIP: "10.10.0.2",
+				IPHeader: crdv1alpha1.IPHeader{
+					Flags:    2,
+					Protocol: 1,
+					TTL:      63,
+				},
+				Length: 84,
+				SrcIP:  "10.10.1.4",
+			},
+			Phase: crdv1alpha1.Succeeded,
+			Results: []crdv1alpha1.NodeResult{
+				{
+					Node: "k8s-node-worker-1",
+					Observations: []crdv1alpha1.Observation{
+						{
+							Action:    crdv1alpha1.ActionForwarded,
+							Component: crdv1alpha1.ComponentSpoofGuard,
+						},
+						{
+							Action:        crdv1alpha1.ActionForwarded,
+							Component:     crdv1alpha1.ComponentForwarding,
+							ComponentInfo: "Output",
+							TunnelDstIP:   "192.168.77.100",
+						},
+					},
+				},
+				{
+					Node: "k8s-node-control-plane",
+					Observations: []crdv1alpha1.Observation{
+						{
+							Action:        crdv1alpha1.ActionReceived,
+							Component:     crdv1alpha1.ComponentForwarding,
+							ComponentInfo: "Classification",
+						},
+						{
+							Action:        crdv1alpha1.ActionDelivered,
+							Component:     crdv1alpha1.ComponentForwarding,
+							ComponentInfo: "Output",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	liveTfwithNoEndpointNodename := crdv1alpha1.Traceflow{
```

Comment:
Yeah live traceflow without source or  destination node name, so gave the name which is used in the code of `traceflow.go` to specify this condition ( instead of giving `liveTfwithNoSourceorDestinationName` thought `EndpointNodeName` suits better)

Reply:
`Node` is a proper noun in K8s representing the entity running Pods. I think using it to represent source and destination of traceflow is confusing.
More importantly, empty source and destination at the same time is not even possible with Traceflow. Did you test this traceflow?